File: IdeoDigitalApi/IdeoDigitalApi/Models/Invoice.cs
namespace IdeoDigitalApi.Models
{
    public class Invoice
    {
        public int Id { get; set; }
        public DateTime InvoiceDate { get; set; }
        public string BusinessName { get; set; }
        public string BusinessAddress { get; set; }
        public string BusinessPhone { get; set; }
        public string CustomerName { get; set; }
        public string CustomerAddress { get; set; }
        public decimal Subtotal { get; set; }
        public decimal Tax { get; set; }
        public decimal Total { get; set; }


        public List<InvoiceItem> InvoiceItems { get; set; }
    }
}

File: IdeoDigitalApp/ideo-digital/src/app/Models/Invoice.ts
import { InvoiceItem } from "./InvoiceItem";

export class Invoice {

    id: number;
    invoiceDate: Date;
    businessName: string;
    businessAddress: string;
    businessPhone: string;
    customerName: string;
    customerAddress: string;
    subtotal: number;
    tax: number;
    total: number;
    invoiceItems: InvoiceItem[];

    constructor() {
        this.id = 0;
        this.invoiceDate = new Date();
        this.businessName = '';
        this.businessAddress = '';
        this.businessPhone = '';
        this.customerName = '';
        this.customerAddress = '';
        this.subtotal = 0;
        this.tax = 18;
        this.total = 0;
        this.invoiceItems = [];
    }
}
